java-checks/src/main/java/org/sonar/java/checks/helpers/ExpressionsHelper.java

```diff
@@ -221,6 +221,15 @@ private static boolean isAssignedToNonSerializable(ExpressionTree expression) {
     } else {
       return Stream.concat(Stream.of(initializer), assignedExpressionStream);
     }
+  }  
+  
+  public static Optional<ExpressionTree> initializedAndAssignedExpression(Symbol symbol) {
```

Comment:
I'm not sure this name is well describing what this method does. What about removing the `AndAssigned` part? Or `initializedAndNotAssignedExpression`?

In addition, maybe we can use `getSingleWriteUsage` from the same file, doing something similar?

Reply:
Thanks, that's make sense to reuse it.